### PR TITLE
Added Leetcode Problem 230 Solution

### LINKED ISSUE #883: [Feature Request]: Leetcode Problem:230
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

[Kth Smallest Element in a BST]
Given the root of a binary search tree, and an integer k, return the kth smallest value (1-indexed) of all the values of the nodes in the tree.

### Use Case

For users to be able to see the solutions and understand it.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Anshika14528**
## Related Issue

closes #883 

## Description
Solution to the leetcode problem 230(kth smallest element in a bst).

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa-solutions/lc-solutions/0000-0099/0017-letter-combinations-of-a-phone-number.md [hunk 1]
```diff
@@ -1,89 +1,104 @@
 ---
-id: letter-combinations-of-a-phone-number
+id: Letter Combinations of a Phone Number
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0017-letter-combinations-of-a-phone-number.md:2]: replace `id: Letter Combinations of a Phone Number` to `id: letter-combinations-of-a-phone-number`

File: dsa-solutions/lc-solutions/0000-0099/0017-letter-combinations-of-a-phone-number.md [hunk 2]
```diff
@@ -1,89 +1,104 @@
...
 title: Letter Combinations of a Phone Number (LeetCode)
-sidebar_label: 0017-Letter Combinations of a Phone Number
+sidebar_label: 0017-Letter-Combinations-of-a-Phone-Number
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0017-letter-combinations-of-a-phone-number.md:4]: replace `sidebar_label: 0017-Letter-Combinations-of-a-Phone-Number` to `sidebar_label: 0017 Letter Combinations of a Phone Number`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code

**Comment by Anshika14528** [comment]
@Ajay-Dhangar sir, by mistake my this issues solutions also got commited in (Added leetcode problem:226 #913) PR 
please can you either give level3 to PR #913 or please give level2 to this issue please

**Comment by Ajay-Dhangar** [comment]
> @Ajay-Dhangar sir, by mistake my this issues solutions also got commited in (Added leetcode problem:226 #913) PR
> please can you either give level3 to PR #913 or please give level2 to this issue please
> My three issues solution got commited in #913 by mistake

okay, don't worry I merge it and assign you level but next time you can focus on it.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**